spec/unit/functions/inline_epp_spec.rb

```diff
@@ -73,6 +75,29 @@
     expect(eval_template("string was: <%= $string %>")).to eq("string was: the string value")
   end
 
+  context "when using Sensitive" do
+    it "returns an unwrapped sensitive value as a String" do
+      expect(eval_and_collect_notices(<<~END)).to eq(["opensesame"])
+        notice(inline_epp("<%= Sensitive('opensesame').unwrap %>"))
+      END
+    end
+
+    it "rewraps a sensitive value" do
+      # note entire result is redacted, not just sensitive part
+      expect(eval_and_collect_notices(<<~END)).to eq(["Sensitive [value redacted]"])
+        notice(inline_epp("This is sensitive <%= Sensitive('opensesame') %>"))
+      END
+    end
+
+    it "can be double wrapped" do
+      catalog = compile_to_catalog(<<~END)
+        notify { 'title':
+          message => Sensitive(inline_epp("<%= Sensitive('opensesame') %>"))
+        }
+      END
+      expect(catalog.resource(:notify, 'title')['message']).to eq('opensesame')
+    end
+  end
```

Comment:
I'm not  100% about the location of these tests, but the second and third tests fail as expected when run against puppet code that doesn't have this change:

```
  1) the inline_epp function when using Sensitive rewraps a sensitive value
     Failure/Error:
       expect(eval_and_collect_notices(<<~END)).to eq(["Sensitive [value redacted]"])
         notice(inline_epp("This is <%= Sensitive('opensesame') %>"))
       END
     
       expected: ["Sensitive [value redacted]"]
            got: ["This is sensitive Sensitive [value redacted]"]
     
       (compared using ==)
     # ./spec/unit/functions/inline_epp_spec.rb:87:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:180:in `block (2 levels) in <top (required)>'
     # ./.bundle/ruby/2.7.0/gems/webmock-3.9.3/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

  2) the inline_epp function when using Sensitive can be double wrapped
     Failure/Error: expect(catalog.resource(:notify, 'title')['message']).to eq('opensesame')
     
       expected: "opensesame"
            got: "Sensitive [value redacted]"
     
       (compared using ==)
     # ./spec/unit/functions/inline_epp_spec.rb:98:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:180:in `block (2 levels) in <top (required)>'
     # ./.bundle/ruby/2.7.0/gems/webmock-3.9.3/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```

Reply:
Yeah, the location is weird, there's 4 places that these changes could be (evaluator, face, inline_epp, and epp) but to actually put _something_ in those places is largely copying the same tests around.